src/Game/System/GalaxyCometScheduler.cpp

```diff
@@ -0,0 +1,456 @@
+    { "CannonFleetGalaxy", COMET_CATEGORY_COIN_100 },
+    { "HellProminenceGalaxy", COMET_CATEGORY_NORMAL },
+    { "OceanPhantomCaveGalaxy", COMET_CATEGORY_COIN_100 },
+    { "CannonFleetGalaxy", COMET_CATEGORY_NORMAL },
+    { "HellProminenceGalaxy", COMET_CATEGORY_COIN_100 },
+    { "OceanPhantomCaveGalaxy", COMET_CATEGORY_NORMAL },
+    { "HellProminenceGalaxy", COMET_CATEGORY_NORMAL },
+    { "CannonFleetGalaxy", COMET_CATEGORY_NORMAL },
+    { "OceanPhantomCaveGalaxy", COMET_CATEGORY_NORMAL },
+    { "HellProminenceGalaxy", COMET_CATEGORY_COIN_100 },
+    { "CannonFleetGalaxy", COMET_CATEGORY_NORMAL },
+    { "HellProminenceGalaxy", COMET_CATEGORY_NORMAL },
+    { "OceanPhantomCaveGalaxy", COMET_CATEGORY_COIN_100 },
+    { "HellProminenceGalaxy", COMET_CATEGORY_COIN_100 },
+    { "OceanPhantomCaveGalaxy", COMET_CATEGORY_NORMAL },
+    { nullptr, 0 },
+};
+
+const GalaxyCometTimePaper * const sGalaxyCometTimerPaper[] = {
+    sCometTimeTableGrandGalaxy1,
+    sCometTimeTableGrandGalaxy2,
+    sCometTimeTableGrandGalaxy3,
+    sCometTimeTableGrandGalaxy4,
+    sCometTimeTableGrandGalaxy5,
+    sCometTimeTableGrandGalaxy6,
+};
+
+const GalaxyCometSerializeInfo overrideTerraceDomeState = { 0x0007, 0 };
+
+const GalaxyCometSerializeInfo firstContact[] = {
+    { 0x0001, 0 },
+    { 0x0000, 0 },
+    { 0x0000, 0 },
+    { 0x0000, 0 },
+    { 0x0000, 0 },
+    { 0x0000, 0 },
+};
+
+GalaxyCometTimeTable::GalaxyCometTimeTable(const GalaxyCometTimePaper *pTimePaper) {
+    mState = nullptr;
+    mTimePaper = nullptr;
+    mTimePaperPos = 0;
+    mIsReady = false;
+    mIsHide = false;
+    mState = new GalaxyCometState();
+    mTimePaper = pTimePaper;
+    mTimePaperPos = 0;
+}
+
+void GalaxyCometTimeTable::update() {
+    if (mIsReady) {
+        mState->update();
+        updateTimePaperPos();
+    }
+}
+
+void GalaxyCometTimeTable::checkIsReady() {
+    s32 pos;
+    bool foundSuitableTimePaperPos = findSuitableTimePaperPos(&pos, -1);
+
+    if (mIsReady != foundSuitableTimePaperPos) {
+        mIsReady = foundSuitableTimePaperPos;
+        if (foundSuitableTimePaperPos) {
+            mTimePaperPos = pos;
+        }
+    }
+}
+
+void GalaxyCometTimeTable::serializeStateToGameData(GalaxyCometSerializeInfo *pInfo) {
+    s32 stateIndex = mState->getStateIndex();
+    pInfo->mStateIndexPacked = (mTimePaperPos << 1) + stateIndex;
+
+    s32 pastSecond = mState->getPastSecond();
+    pInfo->mPastSecond = pastSecond;
+}
+
+void GalaxyCometTimeTable::deserializeStateFromGameData(const GalaxyCometSerializeInfo *pInfo) {
+    u32 stateIndex = pInfo->mStateIndexPacked;
+    u32 stateMask = stateIndex >> 31;
+    u32 timePaperPos = stateIndex >> 1;
+    
+    stateIndex &= 1;
+    stateIndex ^= stateMask;
+    mTimePaperPos = timePaperPos;
+    
+    s32 state = stateIndex - stateMask;
+    mState->setStateAndPastSecond(state, pInfo->mPastSecond);
+}
+
+bool GalaxyCometTimeTable::isLand() const {
+    if (mIsReady == 0) {
```

Comment:
Compare Boolean variables to `true` or `false` as opposed to `0` or `1` if an explicit comparison is needed.